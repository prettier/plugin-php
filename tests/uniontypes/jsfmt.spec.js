run_spec(__dirname, ["php"], { phpVersion: "8.0" });
