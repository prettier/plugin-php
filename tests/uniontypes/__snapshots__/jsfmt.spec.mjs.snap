// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`union.php 1`] = `
====================================options=====================================
parsers: ["php"]
phpVersion: "8.0"
printWidth: 80
                                                                                | printWidth
=====================================input======================================
<?php
function a(int|float $a):int|float {}

class UnionClass {
    public int|float $v = 0;

    public function __construct(public bool|int $flag) {}

    public function get():bool|int|float {}
}

interface UnionInterface { function do(string|int $in):string|bool; }

=====================================output=====================================
<?php
function a(int|float $a): int|float
{
}

class UnionClass
{
    public int|float $v = 0;

    public function __construct(public bool|int $flag)
    {
    }

    public function get(): bool|int|float
    {
    }
}

interface UnionInterface
{
    function do(string|int $in): string|bool;
}

================================================================================
`;
